Update to nodejs rules 0.31.1 & bazel 0.26.0

gregmagolan opened the pull request:
* entry_point attribute of nodejs_binary & rollup_bundle is now a label
* symlinking of node_modules for yarn_install temporarily disabled for bazel schematics and the main yarn_install (not for integration/bazel) until the fix for https://github.com/bazelbuild/bazel/issues/8487 makes it into a future bazel release which will resolve https://github.com/bazelbuild/rules_nodejs/issues/802

gregmagolan committed:
build(bazel): update to nodejs rules 0.30.1 & bazel 0.26.0

nodejs rules 0.30.1 has new feature to symlink node_modules with yarn_install and bazel 0.26.0 includes new managed_directories feature which enables this

kyliau commented on packages/bazel/src/schematics/ng-add/index.ts:
```diff
@@ -48,10 +48,10 @@ function addDevDependenciesToPackageJson(options: Schema) {
     const devDependencies: {[k: string]: string} = {
       '@angular/bazel': angularCoreVersion,
-      '@bazel/bazel': '^0.26.0-rc.5',
+      '@bazel/bazel': '^0.26.0-rc13',
```
rc is not needed since 0.26.0 has been released?

gregmagolan replied:
done. thanks.

gregmagolan commented:
For reference, there was a managed directories flaky crash observed when running the bazel-schematics integration test locally. Issue file to Bazel here https://github.com/bazelbuild/bazel/issues/8488

gregmagolan committed:
build(bazel): update to nodejs rules 0.31.1

* entry_point attribute of nodejs_binary & rollup_bundle is now a label
* symlinking of node_modules for yarn_install temporarily disabled (except for integration/bazel) until the fix for https://github.com/bazelbuild/bazel/issues/8487 makes it into a future bazel release

alexeagle committed:
build: convert entry_point to label

Toxicable commented:
Probably want bazel 0.26.1 with this which fixes managed directories being edited externally (by rm -rf  for example)

alexeagle commented:
Thanks @Toxicable I hadn't seen that the fix was released. Just mirrored that to npm and will update this PR

alexeagle committed:
build: update to Bazel 0.26.1

gregmagolan reviewed (commented):
LGTM. Thanks for finishing it up @alexeagle

IgorMinar reviewed (approved):
I love all the entry-point cleanup! this is so much better!

IgorMinar commented:
mege-assistance: global approval

gregmagolan commented:
Not quite done. Bazel 0.26.1 doesn't fix node_modules issue so need to put back `symlink_node_modules = False` again before merging.

IgorMinar commented:
uh oh.. what's going on with node_modules? why's still broken?

gregmagolan commented:
Scope change on the bazel side getting the managed_directories feature in led to an issue: https://github.com/bazelbuild/rules_nodejs/issues/802 with Bazel 0.26.0. Irina on Bazel team fixed it but hasn't landed in a release yet. Tracking for Bazel team here: https://github.com/bazelbuild/bazel/issues/8487

gregmagolan committed:
build: set symlink_node_modules = False for yarn_install in main WORKSPACE and bazel schematics

This is needed to work-around issue: yarn_install & npm_install with managed directories can't handle deleted or manually regenerated node_modules folder [https://github.com/bazelbuild/rules_nodejs/issues/802]
Underlying issue has been fixed in Bazel https://github.com/bazelbuild/bazel/issues/8487 but hasn't landed in a release yet

gregmagolan commented:
`symlink_node_modules = False` back into main WORKSPACE and Bazel schematics. Ready to merge. These can be removed when Bazel is updated to a release which includes @irengrig fix https://github.com/bazelbuild/bazel/commit/6efc5b787ad3164cc2fb779c73377695032b4524

Toxicable commented:
Ah my mistake, I assumed it was released when I saw the 0.26.1 release along with this being closed
https://github.com/bazelbuild/bazel/issues/8487 at very similar times

IgorMinar commented:
@gregmagolan this one doesn't cleanly land on the patch branch, could you please send a separate PR that targets just 8.0.x? thanks

gregmagolan commented:
> @gregmagolan this one doesn't cleanly land on the patch branch, could you please send a separate PR that targets just 8.0.x? thanks

Roger
